flink-runtime/src/main/java/org/apache/flink/runtime/leaderretrieval/LeaderRetrievalDriver.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.leaderretrieval;
+
+/**
+ * A {@link LeaderRetrievalDriver} is responsible for retrieves the current leader which has been elected by the
+ * {@link LeaderRetrievalDriver}.
+ */
+public interface LeaderRetrievalDriver {
+
+	/**
+	 * Start the necessary services for specific {@link LeaderRetrievalDriver} implementations. For example, NodeCache
+	 * in Zookeeper, ConfigMap watcher in Kubernetes. They could get the leader information update events and need to
+	 * notify the leader listener by {@link DefaultLeaderRetrievalService.StateHandler}.
+	 *
+	 * @param leaderRetrievalStateHandler used for notify the leader changes.
+	 * @throws Exception Throw exception when start the services.
+	 */
+	void start(DefaultLeaderRetrievalService.StateHandler leaderRetrievalStateHandler) throws Exception;
+
+	/**
+	 * Stop the services used for leader retrieval.
+	 */
+	void stop() throws Exception;
```

Comment:
I'd say to call this method `close` instead of `stop`. We could let this interface also extend the `AutoCloseable` interface.